server/src/main/java/org/opensearch/cluster/decommission/DecommissionAttribute.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.decommission;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+public final class DecommissionAttribute implements Writeable {
+    private final String attributeName;
+    private final List<String> attributeValues;
```

Comment:
why are we taking a list here ? Shouldn't we enforce it to be one attribute value ?

Reply:
Kept it as a list for future feature request that might come in to decommission any other node attribute and not just awareness attribute(there might be a use case later on where we might need to decommission multiple awareness attributes as well).

I agree for now, we should just enforce it to one attribute value. And this validation logic can be added in the hook we create to execute this task. So, in future we won't need to update the executor and the responsibility of these checks will lie in the validator

Reply:
Updated the contract, wherein a user can only decommission one zone at a time and request fails if a zone is already decommissioned